engine/src/main/java/org/hibernate/search/engine/metadata/impl/TypeMetadata.java

```diff
@@ -594,15 +606,13 @@ public AnalyzerReference addToScopedAnalyzerReference(DocumentFieldPath fieldPat
 				analyzerReference = scopedAnalyzerReferenceBuilder.getGlobalAnalyzerReference();
 			}
 
-			if ( Field.Index.ANALYZED.equals( index ) || Field.Index.ANALYZED_NO_NORMS.equals( index ) ) {
-				if ( analyzerReference != null ) {
-					scopedAnalyzerReferenceBuilder.addAnalyzerReference( fieldPath.getAbsoluteName(), analyzerReference );
-				}
-			}
-			else {
+			if ( !index.isAnalyzed() ) {
 				// no analyzer is used, add a fake one for queries
```

Comment:
Not related to your patch, but while at it, "fake" is misleading here, we use the pass through analyzer.